tests/std/include/test_regex_support.hpp

```diff
@@ -240,3 +240,64 @@ class test_regex {
         }
     }
 };
+
+class test_wregex {
+    regex_fixture* const fixture;
+    const std::wstring pattern;
+    const std::regex_constants::syntax_option_type syntax;
+    const std::wregex r;
+
+public:
+    test_wregex(regex_fixture* fixture, const std::wstring& pattern,
+        std::regex_constants::syntax_option_type syntax = std::regex_constants::ECMAScript)
+        : fixture(fixture), pattern(pattern), syntax(syntax), r(pattern, syntax) {}
+
+    test_wregex(const test_wregex&) = delete;
+    test_wregex& operator=(const test_wregex&) = delete;
+
+    void should_search_match(const std::wstring& subject, const std::wstring& expected,
+        const std::regex_constants::match_flag_type match_flags = std::regex_constants::match_default) const {
+        std::wsmatch mr;
+        try {
+            const bool search_result = std::regex_search(subject, mr, r, match_flags);
+            if (!search_result || mr[0] != expected) {
+                wprintf(LR"(Expected regex_search("%s", regex("%s", 0x%X), 0x%X) to find "%s", )", subject.c_str(),
+                    pattern.c_str(), static_cast<unsigned int>(syntax), static_cast<unsigned int>(match_flags),
+                    expected.c_str());
+                if (search_result) {
+                    wprintf(LR"(but it matched "%s")"
+                            "\n",
+                        mr.str().c_str());
+                } else {
+                    puts("but it failed to match");
+                }
+
+                fixture->fail_regex();
+            }
+        } catch (const std::regex_error& e) {
+            wprintf(LR"(Failed to regex_search("%s", regex("%s", 0x%X), 0x%X): regex_error: )", subject.c_str(),
+                pattern.c_str(), static_cast<unsigned int>(syntax), static_cast<unsigned int>(match_flags));
+            printf("%s\n", e.what());

```

Comment:
The narrow `test_regex` above printed `e.what()` in quotes. I believe it's preferable for `test_wregex` to emit the same style of output, although they have different ways of forming that output (I like your technique of simply using a narrow `printf` call for the narrow `e.what()` :smiley_cat:):
```suggestion
            printf("\"%s\"\n", e.what());
```